python-lib/query_handlers/scoring_handler.py

```diff
@@ -166,8 +209,14 @@ def _prepare_samples(self):
         cast_mapping = {self.dku_config.users_column_name: "string", self.dku_config.items_column_name: "string"}
         if self.use_explicit:
             cast_mapping[self.dku_config.ratings_column_name] = "double"
+        if self.timestamp_filtering:
+            cast_mapping[self.dku_config.timestamps_column_name] = "date"
         samples_cast = self._cast_table(samples_dataset, cast_mapping, alias="_raw_input_dataset")
-        visit_count = self._build_visit_count(samples_cast)
+        if self.timestamp_filtering:
+            timestamp_filtering = self._build_timestamp_filtered(samples_cast)
+        else:
+            timestamp_filtering = samples_cast
+        visit_count = self._build_visit_count(timestamp_filtering)
```

Comment:
We need to think about the order of the 2 filters.
If we first filter to keep only the last N items per user and then filter items without enough visit, then items with only old visits (even if visited a lot) will be filtered, meaning that we care more about recent visits than highly visited items.
If we first filter items with not enough visit and then keep the last N items per users, then we care more about highly visited items than recent visits.

Reply:
a5b09d3